x-pack/plugins/saved_objects_tagging/public/components/assign_flyout/open_assign_flyout.tsx

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { NotificationsStart, OverlayStart, OverlayRef } from 'src/core/public';
+import { toMountPoint } from '../../../../../../src/plugins/kibana_react/public';
+import { ITagAssignmentService, ITagsCache } from '../../services';
+
+export interface GetAssignFlyoutOpenerOptions {
+  overlays: OverlayStart;
+  notifications: NotificationsStart;
+  tagCache: ITagsCache;
+  assignmentService: ITagAssignmentService;
+  assignableTypes: string[];
+}
+
+export interface OpenAssignFlyoutOptions {
+  /**
+   * The list of tag ids to change assignments to.
+   */
+  tagIds: string[];
+}
+
+export type AssignFlyoutOpener = (options: OpenAssignFlyoutOptions) => Promise<OverlayRef>;
+
+export const getAssignFlyoutOpener = ({
+  overlays,
+  notifications,
+  tagCache,
+  assignmentService,
+  assignableTypes,
+}: GetAssignFlyoutOpenerOptions): AssignFlyoutOpener => async ({ tagIds }) => {
+  const { AssignFlyout } = await import('./assign_flyout');
```

Comment:
nit: It'd be nice if there was some feedback in the UI that this was being loaded. If there was a network delay here, the user wouldn't have any indication that the button worked. We could use a `React.lazy` component in the mount point below with an empty flyout + loading indicator for the fallback component.